File: android-java/app/src/test/java/com/example/dittotasks/DittoModeTest.java
package com.example.dittotasks;

import static org.junit.Assert.assertEquals;

import org.junit.Test;

public class DittoModeTest {
    @Test
    public void nullTokenSelectsOnline() {
        assertEquals(DittoMode.ONLINE_PLAYGROUND, DittoMode.select(null));
    }

    @Test
    public void emptyTokenSelectsOnline() {
        assertEquals(DittoMode.ONLINE_PLAYGROUND, DittoMode.select(""));
    }

    @Test
    public void whitespaceOnlyTokenSelectsOnline() {
        assertEquals(DittoMode.ONLINE_PLAYGROUND, DittoMode.select("   \t\n  "));
    }

    @Test
    public void nonEmptyTokenSelectsOffline() {
        assertEquals(DittoMode.OFFLINE, DittoMode.select("any-real-license-token"));
    }
}

File: android-kotlin/QuickStartTasks/app/src/test/java/live/ditto/quickstart/tasks/DittoModeTest.kt
package live.ditto.quickstart.tasks

import org.junit.Assert.assertEquals
import org.junit.Test

class DittoModeTest {
    @Test
    fun nullTokenSelectsOnline() {
        assertEquals(DittoMode.ONLINE_PLAYGROUND, DittoMode.select(null))
    }

    @Test
    fun emptyTokenSelectsOnline() {
        assertEquals(DittoMode.ONLINE_PLAYGROUND, DittoMode.select(""))
    }

    @Test
    fun whitespaceOnlyTokenSelectsOnline() {
        assertEquals(DittoMode.ONLINE_PLAYGROUND, DittoMode.select("   \t\n  "))
    }

    @Test
    fun nonEmptyTokenSelectsOffline() {
        assertEquals(DittoMode.OFFLINE, DittoMode.select("any-real-license-token"))
    }
}
